src/Checkbox/Checkbox.jsx

```diff
@@ -57,19 +57,11 @@ export const Checkbox = ({
   )
 }
 
-const StyledLabel = styled.label(
+const StyledLabel = styled.label(({ pointerEvents }) =>
   css({
     display: 'flex',
-    alignItems: 'flex-start',
-    justifyContent: 'flex-start',
-    minHeight: '40px',
-    userSelect: 'none'
-  })
-)
-
-const StyledIcon = styled(Icon)(
-  css({
-    minWidth: '24px'
+    userSelect: 'none',
+    pointerEvents
```

Comment:
The label will always receive the click event, so it can be a target for the pointer events. The `for` or `htmlFor` prop on the label will forward the click event on the input checkbox.